src/devices/System.Device.Model/README.md

```diff
@@ -0,0 +1,111 @@
+
+```csharp
+        [Telemetry]
+        public Temperature Temperature => Temperature.FromDegreesCelsius(42.5f + ReadInt16(Register.Temperature) / 480f);
+
+        [Telemetry("Humidity")]
+        public bool TryReadHumidity(out RelativeHumidity humidity) => TryReadHumidityCore(out humidity);
+
+        [Telemetry("Pressure")]
+        public Pressure ReadPressure() { /*...*/ }
+```
+
+- if telemetry is not producing typed unit (i.e. `Vector3`) it should have additional `displayName` provided
+- optional arguments are treated as if they were not there
+- it's not allowed to have more than one `Telemetry` attribute with the same name on the same `Interface`
+
+## PropertyAttribute
+
+Properties should be put on properties or methods which describe the device or change its functioning.
+They should only be used on things which don't change value between calls (unless it's been written to or a command has been executed on the device).
+Specifially reading (telemetry) from the device should not change the state of the property.
+
+Usage is similar to Telemetry with some additions:
+- they can be writable
+  - if same name (i.e. `PowerMode`) is used on i.e. `SetPowerMode` and `ReadPowerMode` they will be merged into a single model property
+- they can be put on methods without return value taking one argument (it must not be be passed by reference)
+- it's not allowed for more than one writers or readers with the same name to be present on the same interface
+
+```csharp
+        [Property]
+        public Sampling HumiditySampling { get { /*...*/ } set { /* ... */ } }
+
+        [Property("PowerMode")]
+        public void SetPowerMode(Bme680PowerMode powerMode) { /*...*/ }
+        [Property("PowerMode")]
+        public Bme680PowerMode ReadPowerMode() { /*...*/ }
+```
+
+## ComponentAttribute
+
+Components represent references to other (instances) of interfaces.
+They can only be put on the properties, the return type of the property or its ancestor class should have an `Interface` attribute.
+
+```csharp
+        [Component]
+        public SenseHatTemperatureAndHumidity TemperatureAndHumidity { get; private set; }
+
+// ...
+    public class SenseHatTemperatureAndHumidity : Hts221.Hts221 { /* ... */ }
+// ...
+    [Interface("HTS221 - Capacitive digital sensor for relative humidity and temperature")]
+    public class Hts221 : IDisposable { /* ... */ }
+```
+
+## CommandAttribute
+
+Commands can be something which can be executed on the device and they can be put only on methods.
+
+```csharp
+        [Command]
+        public void PlayTone(double frequency, int duraton) { /* ... */ }
+        [Command]
+        protected override void SetDefaultConfiguration() { /* ... */ }
+```
+
+## Type serialization
+
+Only simple types can be serialized:
+- enums (without Flags attribute)
+  - values out of enum range are not permitted (i.e. bitwise combination)
+- UnitsNet units
+- basic C# types
+- Vector2, Vector3, Vector4
+- Color
```

Comment:
Maybe this should be fully qualified type names to make it easier to user what we refer to

Reply:
Should we do the switch to portable library like ImageSharp at this point?

Reply:
this doesn't really depend on it yet, once we convert Iot.Device.Bindings to ImageSharp that's when we should deprecate serialization of Color, that's why it's still experimental. Also only serializer will really depend on System.Drawing not the library itself